src/blocks/time.rs

```diff
@@ -79,11 +83,14 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
         let full_time = get_time(format, timezone, locale);
         let short_time = format_short.map(|f| get_time(f, timezone, locale));
 
-        if let Some(short) = short_time {
-            widget.set_texts(full_time, short);
+        let format = if let Some(short) = short_time {
+          FormatConfig { full: Some(full_time.parse()?), short: Some(short.parse()?) }.with_default("")?
```

Comment:
Since `time` is already special in the way it handles formatting, it might be easier to just substitute `$icon` (or `%icon` to be consistent with other "placeholders"?) in in the beginning?

Reply:
I did it like that and left it after "^icon_" support for 2 reasons:

1. `$icon` for consistency with all other blocks, this block is unique enough. I'd probably even prefer all chrono placeholders as "$" than the reverse
2.  Format supports "^icon_", with "set_text" it will no longer be possible, but this most likely won't be that useful here anyway :smile: 

When you say "substitue", you mean to just "get_time(format, timezone, locale).replace("$icon", ...)" and leave it with `set_text`, or something else?

Reply:
I meant something like

```rust
let format = config.format.full.replace("%icon", api.get_icon("time")?);
// same for short

loop {
    // main loop
}
```

It just feels weird to parse format string on every iteration. But I'm not completely against it either.

Reply:
I've simplified the code a little bit but left the parsing in the inner loop because tbh both options sound suboptimal and something like an exception.  

Maybe that's something for https://github.com/greshake/i3status-rust/issues/1605 to consider?

It's a lot of more verbose but aligns with other blocks and placeholders. 

`$timestamp.datetime(" %a %d/%m %R ")` and add Value::datetime + formatter with chrono under the hood.

Reply:
That might work too :)

So let's leave the code as is and deal with this later (can you update your comment on the follow-up work please?)

Reply:
Done! https://github.com/greshake/i3status-rust/pull/1618#issuecomment-1249491109